src/script/lua_api/l_mapgen.cpp

```diff
@@ -389,6 +389,7 @@ Biome *read_biome_def(lua_State *L, int index, const NodeDefManager *ndef)
 	b->heat_point      = getfloatfield_default(L,  index, "heat_point",      0.f);
 	b->humidity_point  = getfloatfield_default(L,  index, "humidity_point",  0.f);
 	b->vertical_blend  = getintfield_default(L,    index, "vertical_blend",  0);
+	b->weight          = getfloatfield_default(L,  index, "weight",          1.f);
```

Comment:
Since we divide by this, it might be good to have a check here which throws an error when this is zero, or clamps it to be at least some epsilon. We also don't want this to be negative.

Reply:
What is the preferred way of doing this in minetest?

Reply:
I added a condition where the value is used.

Reply:
I would personally prefer throwing a `LuaError` if `weight <= 0`, everything else might be surprising. I think the current condition might surprise me since it turns a weight <= 0 into 1.

Reply:
The current code now simply treats <= 0 as "disabled", which is probably most consistent.
The default weight of 1.0 is applied when loading the table from Lua.

Reply:
If by "disabled" you mean "takes no effect" (which is equivalent to a weight of 1), indeed. As said I think being strict (throwing an error for invalid values, which would spare us the check [here](https://github.com/minetest/minetest/pull/15142/files#diff-362450d63f7ccc51c07f0fe80b971a9253120f94bd9a95f1252dcb666c0c0f5cR276)) is usually the best idea for new features (otherwise you get surprises and modders being surprised and/or accidentally relying on them so you don't even have an easy time taking them out later).

But I don't mind this much, I can live with the current code as well.